std/typecons.d

```diff
@@ -5189,6 +5189,8 @@ mixin template Proxy(alias a)
                 return a.opCmp(b);
             else static if (is(typeof(b.opCmp(a))))
                 return -b.opCmp(a);
+            else static if (isFloatingPoint!ValueType || isFloatingPoint!B)
+                return a < b ? -1 : a > b ? +1 : a == b ? 0 : float.nan;
             else
                 return a < b ? -1 : a > b ? +1 : 0;
```

Comment:
Done.

I'm gonna try and have some discussions with compiler writers about opCmp and perhaps do some more optimisation work once I've got a clearer picture of what is/isn't working and why. 

Typedef ought to be as near as 100% of the speed of the original as possible, otherwise it's a pretty awful substitute for a builtin feature.